spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -770,6 +784,10 @@ private void registerChannels() {
 				enabled = smartMatch(this.enabledStatsPatterns, name);
 				if (enabled != null) {
 					monitor.enableStats(enabled);
+					metrics.setFullStatsEnabled(enabled);
```

Comment:
No, I meant that this is a redundant operation, if our monitor isn't `ConfigurableMetricsAware`. We don't populate `metrics` in that case